### PR TITLE
672: Metadata Data Access Layer Interface

### PR DESCRIPTION
**Pull request by halprin**
# Metadata Data Access Layer Interface

What do we think about this interface?  I literally thought about this in 10 seconds.  This could be hot garbage.  What are people thinking?

## Issue

#672.

## Checklist

- [x] I have added JavaDocs where required

**Note**: You may remove items that are not applicable

### CODE REVIEW

**Review by basiliskus — COMMENTED** [review]
Left some comments/questions but it looks good to me!

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetaDataStorage.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package gov.hhs.cdc.trustedintermediary.etor.metadata;
+
+public interface PartnerMetaDataStorage {
+    PartnerMetadata readMetadata(String uniqueId);
```
[the remark this answers is not in the document]
Comment (halprin) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetaDataStorage.java:4]: Exactly.  We have another task in the story to figure out what the heck that ID is.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/DatabasePartnerMetaDataStorage.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package gov.hhs.cdc.trustedintermediary.external.database;
+
+import gov.hhs.cdc.trustedintermediary.etor.metadata.PartnerMetaDataStorage;
+import gov.hhs.cdc.trustedintermediary.etor.metadata.PartnerMetadata;
+
+public class DatabasePartnerMetaDataStorage implements PartnerMetaDataStorage {
```
[the remark this answers is not in the document]
Comment (halprin) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/DatabasePartnerMetaDataStorage.java:6]: Agreed.  I personally prefer `Metadata`.

**Review by jcrichlake — APPROVED** [review]

**Review by basiliskus — APPROVED** [review]

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataStorage.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package gov.hhs.cdc.trustedintermediary.etor.metadata;
+
+/** Interface to store and retrieve our partner-facing metadata. */
+public interface PartnerMetadataStorage {
+    PartnerMetadata readMetadata(String uniqueId);
+
+    /**
+     * This method will do "upserts". If the record doesn't exist, it is created. If the record
```
Comment (JohnNKing) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataStorage.java:8]: Thinking about record add vs update: is there a risk of overwriting metadata that we would need to maintain as part of the official record of submitted orders?

**Pull request merged**